assets/js/googlesitekit/datastore/user/user-input-settings.js

```diff
@@ -0,0 +1,197 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+const { commonActions, createRegistrySelector } = Data;
+
+function fetchStoreReducerCallback( state, inputSettings ) {
+	return {
+		...state,
+		inputSettings: Object
+			.keys( inputSettings )
+			.reduce( ( accum, key ) => ( { ...accum, [ key ]: inputSettings[ key ].values } ), {} ),
+	};
+}
+
+const fetchGetUserInputSettingsStore = createFetchStore( {
+	baseName: 'getUserInputSettings',
+	controlCallback: () => API.get( 'core', 'user', 'user-input-settings', undefined, { useCache: false } ),
+	reducerCallback: fetchStoreReducerCallback,
+} );
+
+const fetchSaveUserInputSettingsStore = createFetchStore( {
+	baseName: 'saveUserInputSettings',
+	controlCallback: ( settings ) => API.set( 'core', 'user', 'user-input-settings', { settings } ),
+	reducerCallback: fetchStoreReducerCallback,
+	argsToParams: ( settings ) => settings,
+	validateParams: ( settings ) => {
+		invariant( isPlainObject( settings ), 'a valid settings are required.' );
+	},
+} );
+
+// Actions
+const SET_USER_INPUT_SETTINGS = 'SET_USER_INPUT_SETTINGS';
+const SET_USER_INPUT_SETTING = 'SET_USER_INPUT_SETTING';
+
+const baseInitialState = {
+	inputSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Sets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} values User input settings.
+	 * @return {Object} Redux-style action.
+	 */
+	setUserInputSettings( values ) {
+		return {
+			type: SET_USER_INPUT_SETTINGS,
+			payload: { values },
+		};
+	},
+
+	/**
+	 * Sets user input setting.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}         settingID Setting key.
+	 * @param {Array.<string>} values    User input settings.
+	 * @return {Object} Redux-style action.
+	 */
+	setUserInputSetting( settingID, values ) {
+		return {
+			type: SET_USER_INPUT_SETTING,
+			payload: {
+				settingID,
+				values,
+			},
+		};
+	},
+
+	/**
+	 * Saves user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} settings User input settings.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*saveUserInputSettings( settings ) {
```

Comment:
Per the ACs, this action should be parameter-less. It should simply save the current settings from the client. In other words, the functionality should match `saveSettings` from `createSettingsStore` - including the need for manually doing a `receiveError` call because this action doesn't require arguments while the internal action does.

Reply:
Updated.